// usrdb/index.js
var Database = require('better-sqlite3');
var db = new Database(__dirname+'/db/aid.db');

// === usr ===
exports.mailUsr=function(email){
var stm=db.prepare('select * from usr where email=?');
var row=stm.get(email);
return row;
};
exports.psslUsr=function(arg2){
var stm=db.prepare('select * from usr where pss=?');
var row=stm.get(arg2);
return row;
};
exports.namUsr=function(arg2){
var stm=db.prepare('select * from usr where name=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};
exports.allUsr=function(){
var stm=db.prepare('select * from usr ');
var row=stm.all();
return row;
};
exports.phnUsr=function(arg2){
var stm=db.prepare('select * from usr where phn=?');
try{var row=stm.get(arg2);
}catch(err){console.log(err);}
return row;
};

// === usr run ===
exports.updUsr=function(uni,usr,sku){
var stm=db.prepare('update usr set uni=? where usr=? and sku=? ');
try{stm.run(uni,usr,sku)}
catch(err){console.log(err)}
};

exports.insUsr=function(name,pss,email){
var stm=db.prepare('insert into usr values(?,?,?) ');
stm.run(name,pss,email)
};

exports.delusr=function(){
var stm=db.prepare('delete from usr where uni=0');
try{stm.run();
}catch(err){var err=err;console.log(err);}
};
// === adr ===
exports.emaAdr=function(ema){
var stm=db.prepare('select * from adr where email=? ');
try{var row=stm.get(ema);}catch(err){console.log(err)}
return row;
};
exports.insAdr=function(email,phn,ln1,ln2,city,sta,zip){
var stm=db.prepare('insert into adr values(?,?,?,?,?,?,?)')
try{stm.run(email,phn,ln1,ln2,city,sta,zip);}
catch(err){console.log(err);}
};
exports.phnAdr=function(phn,email){
var stm=db.prepare('update adr set phn=? where email=?')
    try{stm.run(phn,email);}
    catch(err){console.log(err);}
};

// === pid ===

exports.insPid=function(email,pid,mnt,buy,ite,tim){
var stm=db.prepare('insert into pid values(?,?,?,?,?,?)');
stm.run(email,pid,mnt,buy,ite,tim)
};

exports.getPid=function(email){
var stm=db.prepare('select *,rowid from pid where email=? order by rowid desc ');
try{var row=stm.get(email);
}catch(err){console.log(err);}
return row;
};

exports.allPid=function(email){
var stm=db.prepare('select *,rowid from pid where email=? order by rowid desc ');
try{var row=stm.all(email);
}catch(err){console.log(err);}
return row;
};

exports.selPid=function(pid){
var stm=db.prepare('select *,rowid from pid where email=?');
try{var row=stm.get(pid);
}catch(err){console.log(err);}
return row;
};

exports.maxPid=function(){
var stm=db.prepare('select rowid from pid order by rowid desc');
try{var row=stm.get();
}catch(err){console.log(err);}
return row;
};

exports.idPid=function(id){
var stm=db.prepare('select * from pid where rowid=?');
try{var row=stm.get(id);
}catch(err){console.log(err);}
return row;
};

exports.insQR=function(pid,qr){
var stm=db.prepare('insert into qr values(?,?)');
stm.run(pid,qr);
};

exports.selQR=function(pid){
var stm=db.prepare('select * from qr where pid=?');
try{var row=stm.get(pid);
}catch(err){console.log(err);}
return row;
};


// === pal ===

exports.insPal=function(email,id,mnt,ite,utc){
var stm=db.prepare('insert into pal values(?,?,?,?,?)');

try{stm.run(email,id,mnt,ite,utc)}
catch(err){console.log(err)}
};

exports.itePal=function(ite,tok){
var stm=db.prepare('update pal set ite=? where tok=?');
stm.run(ite,tok);
};

exports.getPal=function(email){
var stm=db.prepare('select *,rowid from pal where email=? order by rowid desc ');
try{var row=stm.get(email);
}catch(err){console.log(err);}
return row;
};

exports.allPal=function(email){
var stm=db.prepare('select *,rowid from pal where email=? order by rowid desc ');
try{var row=stm.all(email);
}catch(err){console.log(err);}
return row;
};

// === fav===

exports.selFav=function(arg2){
var stm=db.prepare('select fst from fav where ema=?');
try{var row=stm.get(arg2);
}catch(err){console.log(err);}
return row;
};

exports.emaFav=function(arg2){
var stm=db.prepare('select * from fav where ema=?');
try{var row=stm.get(arg2);
}catch(err){console.log(err);}
return row;
};

// === fav run ===
exports.setFav=function(fst,ema){
var stm=db.prepare('update fav set fst=? where ema=? ');
try{stm.run(fst,ema);
}catch(err){console.log(err);}
};


exports.delFav=function(email){
var stm=db.prepare('delete from fav where email=?');
try{stm.run(email);
}catch(err){console.log(err);}
}

// === rid ===

exports.insRid=function(email,rid,ite,adr,cre,upd){
var stm=db.prepare('insert into rid values(?,?,?,?,?,?)');
stm.run(email,rid,ite,adr,cre,upd)
};

exports.getRid=function(email){
var stm=db.prepare('select *,rowid from rid where email=? order by rowid desc ');
try{var row=stm.get(email);
}catch(err){console.log(err);}
return row;
};

exports.allRid=function(email){
var stm=db.prepare('select *,rowid from rid where email=? order by rowid desc ');
try{var row=stm.all(email);
}catch(err){console.log(err);}
return row;
};
